File: clojure/pe10.clj
(defn is-prime? [n]
	(let [lim (Math/sqrt n)]
		(loop [x 2]
			(if (> x lim)
				true
				(if (= (mod n x) 0)
					false
					(recur (inc x)))))))

(loop [x 2 sum 0]
  (if (is-prime? x)
    (if (< x 2000000)
      (recur (inc x) (+ sum x))
      (println (str "sum: " sum)))
    (recur (inc x) sum)))

File: ruby/pe10.rb
class Pe10

	def start
		puts Time.new
		@limit = 2000000
		@sublimit = Math.sqrt(@limit) 
		sum = 0
		for i in 2..@limit do
			if test_prime(i)
				sum += i
			end
		end

		puts sum

		puts Time.new
	end

	def test_prime(candidate)
		if candidate > 2
			for i in 2...@sublimit do
				if candidate % i == 0 && candidate != i		
					return false
				end
			end
		end
		true
	end

end

if __FILE__ == $0
	p = Pe10.new
	p.start
end
